core/controller/src/main/scala/org/apache/openwhisk/core/controller/Packages.scala

```diff
@@ -303,9 +319,20 @@ trait WhiskPackagesApi extends WhiskCollectionAPI with ReferencedEntities {
           logging.error(this, s"unexpected package binding refers to itself: $docid")
           terminate(UnprocessableEntity, Messages.packageBindingCircularReference(b.fullyQualifiedName.toString))
         } else {
-          getEntity(WhiskPackage.get(entityStore, docid), Some {
-            mergePackageWithBinding(Some { wp }) _
-          })
+
+          /** Here's where I check package execute only case with package binding. */
+          val packagePath = wp.namespace.asString
+          val bindingPath = wp.binding.iterator.next().toString
```

Comment:
is `wp.binding.iterator` guaranteed to not be empty?

Reply:
Yes, because this code falls within a binding case so binding has to hold some value.

Reply:
ah not enough context shown in the diff; then I think you can replace `wp.binding.iterator.next().toString` with `b.toString` to make this more clear